src/BuildingRegistry/Building/Building_BuildingUnit.cs

```diff
@@ -249,6 +249,57 @@ public void CorrectDeregulationBuildingUnit(BuildingUnitPersistentLocalId buildi
                 .ReplaceAddressAttachmentFromBuildingUnitBecauseStreetNameWasReaddressed(previousAddressPersistentLocalId, newAddressPersistentLocalId);
         }
 
+        public void MoveBuildingUnitInto(
+            Building sourceBuilding,
+            BuildingUnitPersistentLocalId buildingUnitPersistentLocalId,
+            IAddCommonBuildingUnit addCommonBuildingUnit)
+        {
+            sourceBuilding.GuardRemovedBuilding();
+            var buildingUnit = sourceBuilding._buildingUnits.GetNotRemovedByPersistentLocalId(buildingUnitPersistentLocalId);
+            buildingUnit.GuardCommonUnit();
+
+            GuardRemovedBuilding();
+            GuardBuildingValidStatuses(BuildingStatus.Planned, BuildingStatus.UnderConstruction, BuildingStatus.Realized);
+
+            var status =
+                buildingUnit.Status == BuildingStatus.Realized
```

Comment:
Building vs BuildingUnit

Reply:
Done, but actually these implicit operators to string are quite dangerous for these kinds of situations.